DeviceSimulation/SimulationTests.cs

```diff
@@ -1,33 +1,302 @@
+            HttpWebRequest stopSimulationRequest = this.Create_Simulation_Request(simulationContentByteArray);
+            Stream dataStream = stopSimulationRequest.GetRequestStream();
+            dataStream.Write(simulationContentByteArray, 0, simulationContentByteArray.Length);
+            dataStream.Close();
+            var stopSimulationResponse = (HttpWebResponse)stopSimulationRequest.GetResponse();
+
+            //Assert
+            Assert.Equal(HttpStatusCode.OK, stopSimulationResponse.StatusCode);
+
+            var verificationRequest = new HttpRequest(Constants.DS_ADDRESS + "/simulations/1");
+            var verificationResponse = this.httpClient.GetAsync(verificationRequest).Result;
+            JObject jsonResponse = JObject.Parse(verificationResponse.Content);
+
+            Assert.False((bool)jsonResponse["Enabled"]);
+        }
+
+        private string get_ETag_Of_Running_Simulation()
+        {
+            this.Should_Delete_Existing_Simulation();
+            var simulation = JObject.Parse(@"{   
+                'Enabled': true, 
+                'DeviceModels': [   
+                    {   
+                        'Id': 'truck-01', 
+                        'Count': 5 
+                    } 
+                ]
+            }");
+            var createSimulationRequest = new HttpRequest(Constants.DS_ADDRESS + "/simulations");
+            createSimulationRequest.SetContent(simulation);
+            var createSimulationResponse = this.httpClient.PostAsync(createSimulationRequest).Result;
+            Assert.Equal(HttpStatusCode.OK, createSimulationResponse.StatusCode);
+
+            var currentSimulationRequest = new HttpRequest(Constants.DS_ADDRESS + "/simulations/1");
+            var currentSimulationResponse = this.httpClient.GetAsync(currentSimulationRequest).Result;
+            Assert.Equal(HttpStatusCode.OK, currentSimulationResponse.StatusCode);
+            JObject JsonResponse = JObject.Parse(currentSimulationResponse.Content);
+
+            string ETag = (string)JsonResponse["ETag"];
+            return ETag;
+        }
+
+        private void Should_Delete_Existing_Simulation()
+        {
+            var runningSimulationRequest = new HttpRequest(Constants.DS_ADDRESS + "/simulations/1");
+            var runningSimulationResponse = this.httpClient.GetAsync(runningSimulationRequest).Result;
+            
+            //delete current simulation only if it exists
+            if(runningSimulationResponse.StatusCode == HttpStatusCode.OK)
+            {
+                var deleteCurrentSimulationRequest = new HttpRequest(Constants.DS_ADDRESS + "/simulations/1");
+                var deleteCurrentSimulationResponse = this.httpClient.DeleteAsync(deleteCurrentSimulationRequest).Result;
+                Assert.Equal(HttpStatusCode.OK, deleteCurrentSimulationResponse.StatusCode);
+                
+                var getCurrentSimulationRequest = new HttpRequest(Constants.DS_ADDRESS + "/simulations/1");
```

Comment:
same here and other places - SIMULATION_URL  ?

Reply:
http://localhost:9003/v1/simulations/1 - this is the URL that is being used across many tests. So storing it in Constants.cs and reading it from there.